File: core/src/main/java/nl/t64/game/rpg/components/party/SchoolType.java
package nl.t64.game.rpg.components.party;

import lombok.AllArgsConstructor;


@AllArgsConstructor
public enum SchoolType implements SuperEnum {

    NONE("No"),
    UNKNOWN("Unknown"),
    NEUTRAL("Neutral"),
    ELEMENTAL("Elemental"),
    NAMING("Naming"),
    NECROMANCY("Necromancy"),
    STAR("Star");

    final String title;

    @Override
    public String getTitle() {
        return title;
    }

}

File: core/src/nl/t64/game/rpg/components/party/spells/SchoolType.kt
package nl.t64.game.rpg.components.party.spells

import nl.t64.game.rpg.components.party.SuperEnum


enum class SchoolType(override val title: String) : SuperEnum {

    NONE("No"),
    UNKNOWN("Unknown"),
    NEUTRAL("Neutral"),
    ELEMENTAL("Elemental"),
    NAMING("Naming"),
    NECROMANCY("Necromancy"),
    STAR("Star");

}
